### PR TITLE
Update optimization docs

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of docs revamp #7666

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7

### CODE REVIEW

File: docs/source/common/optimizers.rst [hunk 1]
```diff
@@ -64,17 +65,40 @@ Here is a minimal example of manual optimization.
...
 
-Gradient accumulation
----------------------
-You can accumulate gradients over batches similarly to
-:attr:`~pytorch_lightning.trainer.trainer.Trainer.accumulate_grad_batches` of automatic optimization.
-To perform gradient accumulation with one optimizer, you can do as such.
+Access your Own Optimizer
+=========================
+
+``optimizer`` is a :class:`~pytorch_lightning.core.optimizer.LightningOptimizer` object wrapping your own optimizer
```
Comment (tchaton) [inline docs/source/common/optimizers.rst:72]:
```suggestion
The provided ``optimizer`` is a :class:`~pytorch_lightning.core.optimizer.LightningOptimizer` object wrapping your own optimizer
```

**Review by tchaton — APPROVED** [review]
LGTM !

File: docs/source/common/optimizers.rst [hunk 2]
```diff
@@ -468,23 +464,23 @@ Here we add a learning rate warm-up.
...
-        # update params
-        optimizer.step(closure=optimizer_closure)
 
------
+Access your Own Optimizer
+=========================
 
-Access your own optimizer
--------------------------
 ``optimizer`` is a :class:`~pytorch_lightning.core.optimizer.LightningOptimizer` object wrapping your own optimizer
```
Comment (rohitgr7) [inline docs/source/common/optimizers.rst:480]:
```suggestion
The provided ``optimizer`` is a :class:`~pytorch_lightning.core.optimizer.LightningOptimizer` object wrapping your own optimizer
```

**Comment by kaushikb11** [comment]
@rohitgr7 Could you add description to the docs revamp PRs regarding the motivation for the change?

File: docs/source/common/optimizers.rst [hunk 3]
```diff
@@ -1,29 +1,30 @@
...
+*******************
+Manual Optimization
+*******************
+
 For advanced research topics like reinforcement learning, sparse coding, or GAN research, it may be desirable to
 manually manage the optimization process.
 
 This is only recommended for experts who need ultimate flexibility.
-Lightning will handle only precision and accelerators logic.
+Lightning will handle only acceleration, precision and strategy logic.
```
Comment (kaushikb11) [inline docs/source/common/optimizers.rst:27]:
```suggestion
Lightning will handle only accelerator, precision and strategy logic.
```

**Review by kaushikb11 — APPROVED** [review]

File: docs/source/common/optimizers.rst [hunk 4]
```diff
@@ -64,17 +65,40 @@ Here is a minimal example of manual optimization.
...
+
+
+    def training_step(self, batch, batch_idx):
+        optimizer = self.optimizers()
+
+        # `optimizer` is a `LightningOptimizer` wrapping the optimizer.
+        # To access it, do the following.
+        # However, it won't work on TPU, AMP, etc...
+        optimizer = optimizer.optimizer
+        ...
```
Comment (tchaton) [inline docs/source/common/optimizers.rst:91]:
```suggestion

   class Model(LightningModule):

            def __init__(self):
                super().__init__()
                self.automatic_optimization = False
                ...

            def training_step(self, batch, batch_idx):
                optimizer = self.optimizers()
        
                # `optimizer` is a `LightningOptimizer` wrapping the optimizer.
                # To access it, do the following.
                # However, it won't work on TPU, AMP, etc...
                optimizer = optimizer.optimizer
                ...
```

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by Borda — APPROVED** [review]
